### PR TITLE
allow CI to set the Cluster loglevel

### PR DESCRIPTION
**Pull request by aojea**
GCE jobs have log level 4 
https://github.com/kubernetes/kubernetes/blob/17a2e5e990d88cc3f5700d3bfab302b5a2523aba/cluster/gce/config-test.sh#L211-L218

this has performance implications, let's see how much affect

Fixes: https://github.com/kubernetes-sigs/kind/issues/1625

### CODE REVIEW

**Comment by aojea** [comment]
/retest
sig-cli failures

File: hack/ci/e2e-k8s.sh [hunk 1]
```diff
@@ -161,6 +181,10 @@ EOF
     --wait=1m \
     -v=3 \
     "--config=${ARTIFACTS}/kind-config.yaml"
+
+  # Patch kube-proxy to set the verbosity level
```
Comment (BenTheElder) [inline hack/ci/e2e-k8s.sh:185]: eww, we can't do this in kubeadm?
kube-proxy component config?

Comment (aojea) [reply hack/ci/e2e-k8s.sh:185]: I didn't see how, the `command` field is hardcoded
https://github.com/kubernetes/kubernetes/blob/9a20f30745477423e8776bb0d05316c00648f49b/cmd/kubeadm/app/phases/addons/proxy/manifests.go#L78-L81

maybe @neolit123 knows a better way

File: hack/ci/e2e-k8s.sh [hunk 2]
```diff
@@ -161,6 +181,10 @@ EOF
...
+  kubectl patch -n kube-system daemonset/kube-proxy \
+    --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/command/3", "value": "--v='${KIND_CLUSTER_LOG_LEVEL}'" }]'
```
Comment (neolit123) [inline hack/ci/e2e-k8s.sh:187]: `/spec/template/spec/containers/0/command/3`
->
`/spec/template/spec/containers/0/command/-`
to add at the end

File: hack/ci/e2e-k8s.sh (hunk 1 above)
Comment (neolit123) [reply hack/ci/e2e-k8s.sh:185]: the kube-proxy config does not expose klog verbosity. updating klog verbosity *after* a config file is parsed, means one has to do, something like: `flag.Set("v", someValue)`

i don't think any component (that uses componentConfig) supports that yet.

Comment (BenTheElder) [reply hack/ci/e2e-k8s.sh:185]: And kubeadm provides no way to customize kube-proxy args?

Comment (neolit123) [reply hack/ci/e2e-k8s.sh:185]: no, there are no plans to support that.
kubeadm might be deploying a kube-proxy operator in the future:
https://github.com/kubernetes-sigs/cluster-addons/tree/master/kubeproxy

Comment (BenTheElder) [reply hack/ci/e2e-k8s.sh:185]: To clarify: I don't object to the mechanism itself, except that this should really be an initial value, and not involve a restart of all of kube-proxy. _That_ is non-ideal.

I don't think an operator will help this, and if anything will increase the core overhead, I'd love to know more about what motivates this operator.

**Comment by neolit123** [comment]
It should be set initially. Using the kubeadm component config is a no-go,
so the alternative is user patches. This is good idea and we can do it
today, but long term we dont want kubeadm managing kube-proxy component
config upgrades and also such extra container flags overrides could be done
in a custom resource.

The operator is one way of delegating that addon management responsibility
away, but it comes with its own complexities and the configuration flow is
not exactly clear yet.

**Comment by BenTheElder** [comment]
let's try it for now, but I think longterm this will not be the right approach (bash or otherwise).
unblocking, but thinking about alternatives
/lgtm
/approve

File: hack/ci/e2e-k8s.sh [hunk 3]
```diff
@@ -129,6 +129,9 @@ create_cluster() {
     ;;
   esac
 
+# Default Log level for all components in test clusters
+KIND_CLUSTER_LOG_LEVEL=${KIND_CLUSTER_LOG_LEVEL:-4}
```
Comment (BenTheElder) [inline hack/ci/e2e-k8s.sh:133]: TODO: consider switching this to something that matches the rest of CI, if we have it. (kube-up?)

Comment (aojea) [reply hack/ci/e2e-k8s.sh:133]:
```
hack/local-up-cluster.sh:LOG_LEVEL=${LOG_LEVEL:-3}
```
 or
```
cluster/gce/config-test.sh:TEST_CLUSTER_LOG_LEVEL=${TEST_CLUSTER_LOG_LEVEL:---v=4}
```

**Comment by BenTheElder** [comment]
(we have a proposal out about CI issues, not going to block debugging things on waiting for that to be fixed)

**Pull request merged**